prov/gni/src/gnix_fabric.c

```diff
@@ -348,7 +348,13 @@ static int gnix_getinfo(uint32_t version, const char *node, const char *service,
 		if (gnix_info->fabric_attr) free(gnix_info->fabric_attr);
```

Comment:
Looks good, but as a sidenote you don't need to have branches around all of those. You can just do something like:
```
err:
	if (gnix_info) {
		free(gnix_info->tx_attr);
		free(gnix_info->rx_attr);
		free(gnix_info->ep_attr);
        free(gnix_info->domain_attr);
		free(gnix_info->fabric_attr);
		free(gnix_info);
	}

	/*
	 *  for the getinfo method, we need to return -FI_ENODATA  otherwise
	 *  the fi_getinfo call will make an early exit without querying
	 *  other providers which may be avaialble.
	 */
	return -FI_ENODATA;
```